Wangscape/tilegen/CoordinatePacker.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+#include <assert.h>
+
+namespace tilegen
+{
+
+template<typename I>
+class CoordinatePacker
+{
+public:
+    CoordinatePacker(I size_);
```

Comment:
There should be at least `is_arithmetic` type_trait for `I` as you're comparing it with numbers and using `operator*=` and `operator+=` with `I`-typed members.